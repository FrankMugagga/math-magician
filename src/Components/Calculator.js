import React, { useState } from 'react';
// import Child from './ChildComponent';
import calculate from '../logic/calculate';

function Calculator() {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const newData = calculate(calculatorData, buttonName);
    setCalculatorData(newData);
  };

  const displayValue = calculatorData.next || calculatorData.total || '0';

  return (
    <div id="calBody">
      <div className="container">
        <div className="display">{displayValue}</div>

        <div className="keypad">
          <button type="button" className="btn" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" className="btn" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" className="btn" onClick={() => handleClick('%')}>%</button>
          <button className="operatorBtn" type="button" onClick={() => handleClick('รท')}>&divide;</button>
          <button type="button" className="btn" onClick={() => handleClick('7')}>7</button>
          <button type="button" className="btn" onClick={() => handleClick('8')}>8</button>
          <button type="button" className="btn" onClick={() => handleClick('9')}>9</button>
          <button className="operatorBtn" type="button" onClick={() => handleClick('x')}>x</button>
          <button type="button" className="btn" onClick={() => handleClick('4')}>4</button>
          <button type="button" className="btn" onClick={() => handleClick('5')}>5</button>
          <button type="button" className="btn" onClick={() => handleClick('6')}>6</button>
          <button className="operatorBtn" type="button" onClick={() => handleClick('-')}>-</button>
          <button type="button" className="btn" onClick={() => handleClick('1')}>1</button>
          <button type="button" className="btn" onClick={() => handleClick('2')}>2</button>
          <button type="button" className="btn" onClick={() => handleClick('3')}>3</button>
          <button className="operatorBtn" type="button" onClick={() => handleClick('+')}>+</button>
          <button className="btn0" type="button" onClick={() => handleClick('0')}>0</button>
          <button type="button" className="btn" onClick={() => handleClick('.')}>.</button>
          <button className="operatorBtn" type="button" onClick={() => handleClick('=')}>=</button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
