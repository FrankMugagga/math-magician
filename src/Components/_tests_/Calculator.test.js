import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';


