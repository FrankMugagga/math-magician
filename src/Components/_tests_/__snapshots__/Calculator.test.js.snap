// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  id="calBody"
>
  <div
    className="calHead"
  >
    Let's do some math!\`
  </div>
  <div
    className="container"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="keypad"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator component renders correctly with specific props 1`] = `
<div
  id="calBody"
>
  <div
    className="calHead"
  >
    Let's do some math!\`
  </div>
  <div
    className="container"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="keypad"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operatorBtn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
